dot/network/service.go

```diff
@@ -277,16 +277,13 @@ func (s *Service) Start() error {
 		logger.Warnf("failed to register notifications protocol with transaction id %s: %s", transactionsID, err)
 	}
 
-	// since this opens block announce streams, it should happen after the protocol is registered
-	// NOTE: this only handles *incoming* connections
-	s.host.p2pHost.Network().SetConnHandler(s.handleConn)
-
 	// this handles all new connections (incoming and outgoing)
 	// it creates a per-protocol mutex for sending outbound handshakes to the peer
 	s.host.cm.connectHandler = func(peerID peer.ID) {
 		for _, prtl := range s.notificationsProtocols {
 			prtl.peersData.setMutex(peerID)
 		}
+		s.host.cm.peerSetHandler.Incoming(0, peerID)
```

Comment:
1. Adding the `setID` constant since the interface doesn't have named arguments (sus) and it's hard to know what the first arg is.
2. Re-adding the older TODO comment
3. Don't we need the extra logic from the now removed `handleConn`?

```suggestion
		// TODO: currently we only have one set so setID is 0, change this once we have more set in peerSet.
		const setID = 0
		s.host.cm.peerSetHandler.Incoming(setID, peerID)
```

Reply:
It seems that the extra logic in handleConn is handled somewhere else, when ConnHandler interface was removed.  Testing with 3 node devnet confirms that nodes are connecting and communicating.